### DISCUSSION TITLE
How to delay the data rendering from a list in my store?

### DISCUSSION THREAD

**Discussion by waldoc89** [quasarframework/quasar#13864] [General - Components / Directives / etc]
How to delay the data rendering that is fetched via store? 

So this is my component: 
```
        <q-table
            title="Logs"
            separator="horizontal"
            v-model:pagination="pagination"
            dense
            :grid="$q.screen.xs"
            :rows="rows"
            :columns="columns"
            row-key="name"
            hide-bottom
        >
``` 
as you can see, `rows` is a variable generated in the `<script>` section like this: 
```
export default {
    setup() {
        let logs = useLogs();
        let rows = logs.get();
        let clear = () => logs.clear();

        const pagination = ref({
            sortBy: "ts",
            descending: true,
            page: 1,
            rowsPerPage: 50,
            // rowsNumber: 100,
        });
        return {
            columns,
            rows,
            pagination,
            clearLogs: () => (rows = clear()),
        };
    },
};
``` 

log store composable: 
```
const useLogs = (ts, action, status, data) => {
    const store = useStore();

    const push = (ts, action, status, data) => {
        try {
            store.commit("logs/pushToLogs", { ts, action, status, data });
        } catch (e) {
            console.error(`ERROR[useCreateLogMsg] - ${e}`);
        }
    };

    const get = () => store.getters["logs/getLogs"];

    const clear = () => store.commit("logs/clearLogs");

    return { push, get, clear };
};
``` 
in logs store mutation: 
```
const pushToLogs = (state, payload) => {
	return state.push(payload);
};
``` 

Sometimes I get a bunch of data from the store where my browser slow down or crash. 
I used to have a debouncer method on my app, but since im using a framework, i wish to handle the situation in a more framework-ish way. 
How do I delay the rendering from the store? preferably loading one line at a time (1sec for example). just to make this readable by a user.
ps. its important to be able to flag on/off the delay.

**Comment by MilosPaunovic** [comment]
Perhaps this is what you're looking for:
[https://quasar.dev/quasar-utils/other-utils#debounce-function](https://quasar.dev/quasar-utils/other-utils#debounce-function)

**Reply by waldoc89** [reply]
> Perhaps this is what you're looking for: https://quasar.dev/quasar-utils/other-utils#debounce-function

Thanks, I tried applying it directly in the method where i push my data to the store so i can decide maybe to add a flag in the gui (enable/disable it). and I prefere delaying the insertion per-row rather than getting chunks at the display. 

This is a part of a composable function that fetches read-event-triggered data from a bluetooth device and push the data to the store. 
for some reason applying this doesn't do the job and the data gets pushed regularly as if it's not there. 

the main device composable: 
```
const useDevice = (device1, device2) => {
	const connect = async () => {
		switch (device2.connectionType.value) {
			case "bluetooth":
				try {
					connection.value = await useBluetooth(isConnected);
					type.value = "bluetooth";
					watch(
						() => connection.value.read, // <-- watch this
						getRead // <-- trigger this
					);
					return true;
				} catch (e) {
... 


	// Push to Logs
	const pushToLogs = (date, action, status, dataView) =>
		logs.push(Date.now(), "read", "success", parser.get(dataView));

	// Read
	const getRead = (dataView) => {
		read.value = dataView;
		debounce(pushToLogs(Date.now(), "read", "success", dataView), 10000);
		return dataView;
	};
```

i intentionally exaggerated the time because i wasn't sure if it's working or not. 
What am i doing wrong? 

I tried doing this like this too: 
```
		debounce(
			() => pushToLogs(Date.now(), "read", "success", dataView),
			5000
		);
```
or
```
		() =>
			debounce(pushToLogs(Date.now(), "read", "success", dataView), 5000);
```

but this didn't add a new line at all in the store array which i am pushing to (so no new data is displayed). 


I tried adding this to my useBluetooth composable: 
```
const useBluetooth = async (isConnected) => {
	let read = ref(""); // Last read data set by device
	// Read listener
	ch.addEventListener(
		"characteristicvaluechanged",
		debounce((dataView) => {
			read.value = dataView.target.value;
		}, 3000)
	);
	await ch.startNotifications();
...
}
```

and **this works** for single reads, which is nice. but doesn't work when I receive a stream of data - which is what it's meant for. + I prefer having the debounce while inserting the data to the database rather when receiving it.
